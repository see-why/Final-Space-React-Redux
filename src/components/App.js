import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Switch, Route } from 'react-router-dom';
import MainCard from './mainCard';
import { loadcharacters } from '../redux/characters/characters';
import { loadepisodes } from '../redux/episodes/episodes';
import CharacterDetails from './characterDetails';

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loadcharacters());
    dispatch(loadepisodes());
  }, []);

  return (
    <Switch>
      <Route exact path="/">
        <MainCard />
      </Route>
      <Route path="/character/:id" component={CharacterDetails} />
    </Switch>
  );
}

export default App;
