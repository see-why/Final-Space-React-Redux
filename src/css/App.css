a,
.character-Container,
.my-character,
.my-character-images,
.my-character-info,
.my-profile,
.other-info h3,
.span-div,
.abilities-div,
.other-info {
  display: flex;
}

.my-profile {
  background: #ec4c8a;
}

.my-profile,
.character-input {
  margin-bottom: 5px;
}

.fas,
.far {
  right: 5px;
  top: 10px;
  position: absolute;
}

.my-character-info h2,
.my-character-info h3 {
  text-align: end;
}

h3 i.far {
  position: relative;
  top: 4px;
  margin-left: 10px;
}

.zigZag {
  background: #b13967;
}

.abilities {
  flex-direction: column;
}

.abilities-div {
  justify-content: space-between;
}

@media only screen and (max-width: 990) {
  .my-profile-info h3,
  .my-character-info h2,
  .my-character-info h3 {
    font-size: large;
  }
}
