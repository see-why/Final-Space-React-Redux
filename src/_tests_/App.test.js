import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { act } from 'react-dom/test-utils';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../components/App';
import store from '../redux/configureStore';

it('App renders the UI as expected', () => {
  let tree;
  act(() => {
    tree = render(
      <React.StrictMode>
        <Router>
          <Provider store={store}>
            <App />
          </Provider>
        </Router>
      </React.StrictMode>,
    );
  });
  expect(tree).toMatchSnapshot();
});
