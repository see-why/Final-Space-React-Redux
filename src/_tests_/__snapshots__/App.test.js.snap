// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the UI as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="parent-div"
      >
        <div
          class="my-profile"
        >
          <div
            class="slideshow-container"
          >
            <div
              class="mySlides fade"
              style="display: flex;"
            >
              <img
                alt="the_happy_place"
                src="https://finalspaceapi.com/api/episode/image/the_happy_place.jpg"
              />
            </div>
            <div
              class="mySlides fade"
              style="display: none;"
            >
              <img
                alt="the_other_side"
                src="https://finalspaceapi.com/api/episode/image/the_other_side.jpg"
              />
            </div>
            <div
              class="mySlides fade"
              style="display: none;"
            >
              <img
                alt="the_arachnitects"
                src="https://finalspaceapi.com/api/episode/image/the_arachnitects.jpg"
              />
            </div>
            <div
              class="mySlides fade"
              style="display: none;"
            >
              <img
                alt="the_remembered"
                src="https://finalspaceapi.com/api/episode/image/the_remembered.jpg"
              />
            </div>
            <div
              class="mySlides fade"
              style="display: none;"
            >
              <img
                alt="descent_into_darkness"
                src="https://finalspaceapi.com/api/episode/image/the_descent_into_darkness.jpg"
              />
            </div>
          </div>
          <div
            class="my-profile-info"
          >
            <h1>
              FINAL SPACE
            </h1>
            <h3>
              0 
              Characters
            </h3>
            <h3>
              0 
              Episodes
            </h3>
          </div>
        </div>
        <div>
          <input
            class="character-input"
            name="character"
            placeholder="search character name..."
            type="text"
            value=""
          />
          <div
            class="character-Container"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="parent-div"
    >
      <div
        class="my-profile"
      >
        <div
          class="slideshow-container"
        >
          <div
            class="mySlides fade"
            style="display: flex;"
          >
            <img
              alt="the_happy_place"
              src="https://finalspaceapi.com/api/episode/image/the_happy_place.jpg"
            />
          </div>
          <div
            class="mySlides fade"
            style="display: none;"
          >
            <img
              alt="the_other_side"
              src="https://finalspaceapi.com/api/episode/image/the_other_side.jpg"
            />
          </div>
          <div
            class="mySlides fade"
            style="display: none;"
          >
            <img
              alt="the_arachnitects"
              src="https://finalspaceapi.com/api/episode/image/the_arachnitects.jpg"
            />
          </div>
          <div
            class="mySlides fade"
            style="display: none;"
          >
            <img
              alt="the_remembered"
              src="https://finalspaceapi.com/api/episode/image/the_remembered.jpg"
            />
          </div>
          <div
            class="mySlides fade"
            style="display: none;"
          >
            <img
              alt="descent_into_darkness"
              src="https://finalspaceapi.com/api/episode/image/the_descent_into_darkness.jpg"
            />
          </div>
        </div>
        <div
          class="my-profile-info"
        >
          <h1>
            FINAL SPACE
          </h1>
          <h3>
            0 
            Characters
          </h3>
          <h3>
            0 
            Episodes
          </h3>
        </div>
      </div>
      <div>
        <input
          class="character-input"
          name="character"
          placeholder="search character name..."
          type="text"
          value=""
        />
        <div
          class="character-Container"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
